core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -212,5 +227,68 @@ public T shardSize(Integer shardSize) {
             this.shardSize = shardSize;
             return (T)this;
         }
+
+        @Override
+        public String getWriteableName() {
+            return name;
+        }
+
+        @Override
+        public final T readFrom(StreamInput in) throws IOException {
+            T builder = doReadFrom(in);
+            builder.setName(in.readString());
```

Comment:
I think the sub-type (e.g. TermSuggestionBuilder) should read the name and use the constructor).